pkg/proxy/healthcheck/service_health.go

```diff
@@ -36,6 +36,12 @@ import (
 	utilproxy "k8s.io/kubernetes/pkg/proxy/util"
 )
 
+const (
+	// ToBeDeletedTaint is a taint used by the CLuster Autoscaler before marking a node for deletion. Defined in
+	// https://github.com/kubernetes/autoscaler/blob/e80ab518340f88f364fe3ef063f8303755125971/cluster-autoscaler/utils/deletetaint/delete.go#L36
+	ToBeDeletedTaint = "ToBeDeletedByClusterAutoscaler"
```

Comment:
this can be a slippery slope if we have to support all the projects that add taints on nodes :/

Reply:
Hmm, I wanted to start off by saying: "I can't say anything about the future for the proxy code, but the CCM has been looking at only this one taint on services for...", and while looking at the service controller code and inspecting who added this taint, it seems it was added with: https://github.com/kubernetes/kubernetes/pull/105946 which had a very dubious reason for adding it in the first place:

> Prior to https://github.com/kubernetes/kubernetes/pull/90823, these nodes would get removed from the LoadBalancer backend once they were marked as Unschedulable. Now that Unschedulable nodes are tolerated by the node predicate, these nodes will only get removed when the nodes have been deleted.  

Which is not true _today_, the unschedulable predicate was re-added by me in the CCM...but now I am asking myself questions as to whether that was good, the PR removing the unschedulability predicate (#90823) mentioned a production outage

> This caused a production outage when nodes in an autoscaling group were marked unschedulable so the autoscaler would gracefully terminate them - instead the ingress controllers on those nodes were immediately removed from the LB pool and no new ingress requests could be accepted.

/cc @thockin

Reply:
I've revisited this condition as a part of my work on https://github.com/kubernetes/kubernetes/pull/111658. I think we can remove this condition. It didn't have any clear reasons for being added in the first place and logically, today, it doesn't make sense. 

We can discuss this on the on-going KEP I have: https://github.com/kubernetes/enhancements/pull/3460 now during the 1.26 KEP cycle. I thought upon the life cycle of the Node and its relation to the LB in that document. I think we can remove this condition altogether at this point, but I will leave the PR as-is until we get some clarity and consensus on the KEP. Once done, I will update the PR for 1.26.